### PR TITLE
Fix dashboard switching on Mobile

### PR DESCRIPTION
**Pull request by Gusted**
- This is a regression of improving mobile experience on Gitea, currently organization dashboard aren't readable and the popup won't show up when you want to switch between users/organization(as we saw in #19978). 
- This patch fixes that, by allowing the popup to allocate the required pixels(for some absurd reason, z-index doesn't work on the popup, so it's not able to render over the existing elements, we can investigate later of why this is). And also remove the additional dropdown menu for the pages link, so it's one unified list which then can be displayed as rows.

### Before
<details></details>
<details></details>
<details></details>

### After
<details></details>
<details></details>

### CODE REVIEW

File: templates/user/dashboard/navbar.tmpl [hunk 1]
```diff
@@ -75,30 +75,28 @@
 		{{end}}
 
 	{{if .ContextUser.IsOrganization}}
-		<div class="right stackable menu">
```
Comment (zeripath) [inline templates/user/dashboard/navbar.tmpl]: This doesn't seem related to your fix?

(Or your screenshots aren't showing it)

File: web_src/less/_base.less [hunk 2]
```diff
@@ -2194,7 +2194,7 @@ table th[data-sortt-desc] {
   .ui.stackable.menu:not(.no-vertical-tabs) {
     overflow-y: hidden;
     overflow-x: auto;
-    flex-direction: row;
+    flex-direction: row !important;
```
Comment (zeripath) [inline web_src/less/_base.less]: Agh do we really need important here?

**Comment by zeripath** [comment]
Hmm...

Interesting diff, you've removed the "right stackable menu" wrapping div from around the organisations action menu  and then added an !important selector to stackable? These don't seem intimately related to fixing the context selector drop-down and your screenshots don't appear to be showing what happens to this action menu.

File: web_src/less/_base.less (hunk 2 above)
Comment (Gusted) [reply web_src/less/_base.less]: Depends, do we remove the dropdown(currently the case). Then this can be removed. Otherwise this stays 🤷🏽.

**Comment by Gusted** [comment]
Removed the for now unnecessary `!important`.

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
I think we need to do a similar thing to the navbar trick but this is a definite improvement.

There's an issue with @mediaMd spilling over now though.

File: templates/user/dashboard/navbar.tmpl (hunk 1 above)
Comment (zeripath) [inline templates/user/dashboard/navbar.tmpl]: ~~Actually I think this needs to be `<div class="menu">` as without it you don't get the color highlight to say which view is being used.~~

Nope the color is coming from an explicit dependency on right stackable menu.... We need to change dashboard less too.

Comment (Gusted) [reply templates/user/dashboard/navbar.tmpl]: What about changing to `<div class="org-menu">`? And then change the CSS according to that class?

**Comment by 6543** [comment]
what's the current status ?!?

**Comment by Gusted** [comment]
> what's the current status ?!?

Waiting on answer from @zeripath, but I can go ahead and implement my mentioned fix. However https://github.com/go-gitea/gitea/pull/20393 was merged. So we might re-use that logic for the dropdown menu.

**Comment by zeripath** [comment]
Sorry @Gusted didn't realise you were waiting on me - Do you have an updated PR that you were thinking on showing me?

**Comment by Gusted** [comment]
> Do you have an updated PR that you were thinking on showing me?

Currently not. Because I don't have a clue how to proceed. I could see if tippy.js would work for the dropdowns and have a general fix. Or just fiddle around with CSS and HTML structures so the original colors won't be lost, as noted in your review.

**Comment by Gusted** [comment]
Ugh dropdown seems to be bit more complicated to make it compatible with tippy.js. So for now just dirty CSS to fix this issue.

File: web_src/less/_dashboard.less [hunk 3]
```diff
@@ -81,6 +81,12 @@
 
     .ui.dropdown {
       max-width: 100%;
+
+      @media @mediaSm {
+        > .menu:not(.hidden) {
```
Comment (wxiaoguang) [inline web_src/less/_dashboard.less]: Is the `:not(.hidden)` selector necessary? Otherwise L-G-T-M

Comment (Gusted) [reply web_src/less/_dashboard.less]: IIRC it was to have selector priority.

Comment (wxiaoguang) [reply web_src/less/_dashboard.less]: On my side, `> .menu { position: static; } ` works fine (ps: `static` seems more consistent than `inherit`)

```
      @media @mediaSm {
        > .menu {
          position: static;
        }
      }
```


image

**Review by wxiaoguang — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
:rock:
